src/HTTP/UrlParser.cpp

```diff
@@ -198,3 +199,19 @@ std::pair<bool, AString> cUrlParser::Parse(
 
 
 
+bool cUrlParser::Validate(const AString & a_Url)
+{
+	// https://stackoverflow.com/questions/5620235/cpp-regular-expression-to-validate-url
+	// I didn't use the first solution deliberately because it didn't enforce the http(s) in the beginning
+
+	const std::regex UrlRegex (
```

Comment:
Do we really want to pull in the whole regex machinery just to check a simple URL? Not to mention that it's possible that this check says OK and the `Parse` method still fails, or vice versa. I'd just call `Parse` and decide upon its return value.

Reply:
Indeed: The `Parse` function will expect us to provide a protocol. This will work without protocol

Reply:
My idea was, because we need to supply a bunch of String objects for the `Parse` method. This is just a simple take the string and tell me.

Reply:
It is simple, but it needn't match whatever `Parse` is expecting. My point is, keep this function, but rather than using a regex, call `Parse` from here, discarding its outputs other than the error flag. That way you know that whatever this accepts, `Parse` will accept, too.